tmk_core/common/eeconfig.c

```diff
@@ -31,6 +35,9 @@ __attribute__((weak)) void eeconfig_init_kb(void) {
 void eeconfig_init_quantum(void) {
 #ifdef STM32_EEPROM_ENABLE
     EEPROM_Erase();
```

Comment:
should this not be hidden behind `eeprom_driver_erase`, relocating `EEPROM_Erase` to `eeprom_stm32.c`?

Reply:
`eeprom_driver_erase()` is implemented by the implementations in `drivers/eeprom`.
That area does not currently wrap the vendor-provided implementations, and as such was left split up.

Ideally a later PR can refactor this to use the `eeprom_driver_xxxxxxx` interface for the vendor-provided implementations as well, but after discussion on discord it was deemed "safer" to defer to a later PR.